packages/node-opcua-end2end-test/test/end_to_end/test_e2e_issue896_execute_method_access_control.js

```diff
@@ -0,0 +1,89 @@
+"use Strict";
+
+const opcua = require("node-opcua");
+const should = require("should");
+
+describe("Issue 896: Check Authorization for UAMethods", () => {
+    const users = [
+        {username: "Gandalf", password: "g", role: "admin"},
+        {username: "Frodo", password: "f", role: "guest"},
+    ];
+
+    const port = 2003;
+    const server = new opcua.OPCUAServer({
+        port,
+        userManager: {
+            getUserRole: username => users.find(user => user.username === username).role,
+            isValidUser(username, password) {
+                const user = users.find(user => user.username === username);
+                if (!user) return false;
+                return user.password === password;
+            }
+        }
+    })
+
+    const client = opcua.OPCUAClient.create({
+        endpoint_must_exist: false,
+    });
+
+    let wasExecuted;
+
+    before(async () => {
+        await server.initialize();
+        const addressSpace = server.engine.addressSpace;
+        const namespace = addressSpace.getOwnNamespace();
+        const folder = namespace.addFolder(addressSpace.rootFolder.objects, {
+            browseName: "e2e",
+            nodeId: "ns=1;s=e2e"
+        });
+        namespace.addMethod(folder, {
+            browseName: "doIt",
+            nodeId: "ns=1;s=doIt",
+            permissions: {
+                Execute: ["!*", "admin"]
+            }
+        }).bindMethod((inputArguments, context, callback) => {
+            wasExecuted = true;
+            callback(null, {statusCode: opcua.StatusCodes.Good})
+        });
+        await server.start();
+        return client.connect(`opc.tcp://localhost:${port}/UA/NodeOPCUA`);
+    });
+
+    after(() => {
```

Comment:
after(async()=>{
   await client.disconnect();
   await server.shutdown();

});

Reply:
This should fix the test

Reply:
thx